### ISSUE TITLE
Traps need "trigger by attacking" interaction mode pretty desperately.

### ISSUE THREAD

**Issue by Tamior** [CleverRaven/Cataclysm-DDA#47024]
# Is your feature request related to a problem? Please describe.
TL;DR: Most types of conventional traps should trigger if you attack it on purpose.
I.e. if there is a bear trap, you should absolutely be able to trigger it by hitting with a long stick or something.
Right now the _only_ thing you can do about a bear trap in a corridor is either try to sidestep it or try to disarm it. And if you fail to disarm, you get hit by it. 
Obviously, it's almost entirely ridiculous to be hit by a bear trap you are trying to disarm if you have any kind of long heavy implement to trigger it with.
Similar situation applies to most types of conventional traps (you are aware of). You can literally kill yourself by trying to "disarm" a tripwire and instead hitting the floor face-first on and on.

# Describe the solution you'd like
Basically, you can go about solving this issue in a number of ways.
But, in my opinion at least, the most flexible and "natural" solution here would be to allow traps to be triggered by attacking. 
Some basic rules to the effect of "attack should deal at least X damage to trigger the trap" and "trap deals it's damage to the weapon used" should probably apply.

# Describe alternatives you've considered
Adding a new interaction mode "use X to trigger a trap". Seems more complicated and less intuitive. 

# Additional context
I realize full well that this is probably easier said that done. The game does not really support attacking of object via normal attack action as far as I know, and "s"mashing has it's own problems (not being able to use a reach weapon to smash remotely, for one).

**Comment by Night-Pryanik** [comment]
> Right now the only thing you can do about a bear trap in a corridor is either try to sidestep it or try to disarm it.

No, it's not. You can also trigger some traps (bear trap included) by throwing something at it. For bear trap this "something" must be >200 grams (as defined in `trigger_weight` field of the trap). So we already do have some sort of "trigger by attacking" interaction mode.

The only thing that might need an improvement here is _probably_ providing better UI information about `trigger_weight` of traps. Player might guess that he can throw something at bear trap to trigger it, and throw his butter knife, for example. With 50 grams of weight, smartphone won't trigger the trap, and player could think that "trigger-by-throwing" mechanics either is broken or isn't implemented at all.

**Comment by Night-Pryanik** [comment]
Closing as stale, since stalebot can't do this by itself.
